plugin.cpp

```diff
@@ -204,8 +209,19 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
 					return NULL;
 				}
 			}
+			else if (actionStr == "remove")
+			{
+				actn = action::REMOVE;
+
+                                if (iter->HasMember("datapoint"))
+                                        datapoint = (*iter)["datapoint"].GetString();
+
+                                if (iter->HasMember("type"))
+                                        dpType = (*iter)["type"].GetString();
+
+			}
 			Logger::getLogger()->info("Parse asset filter config, Adding to assetFilterConfig map: {%s, %d, %s}", asset_name.c_str(), actn, new_asset_name.c_str());
-			(*info->assetFilterConfig)[asset_name] = {actn, new_asset_name, dpmap};
+			(*info->assetFilterConfig)[asset_name] = {actn, new_asset_name, dpmap, datapoint, dpType};
```

Comment:
why assetFilterConfig has been taken as a pointer? Will simple object will not work instead of a pointer ?

Reply:
base code , did not tamper with it.